src/main/scala/org/renci/cam/QueryService.scala

```diff
@@ -345,21 +357,18 @@ object QueryService extends LazyLogging {
 
   private def getSlotStuff(predicateMap: List[String]): RIO[ZConfig[AppConfig] with HttpClient, List[SlotStuff]] =
     for {
-      values <- Task.effect(
-        predicateMap.zipWithIndex
-          .map(a => String.format(" ( <%s> \"e%s\" ) ", a._1, StringUtils.leftPad(a._2.toString, 4, '0')))
-          .mkString(" "))
-      queryText = s"""SELECT DISTINCT ?qid ?kid ?blslot ?label WHERE {
-          |VALUES (?kid ?qid) { $values }
-          |?kid <http://www.w3.org/2000/01/rdf-schema#subPropertyOf>+ ?blslot .
-          |?blslot a <https://w3id.org/biolink/biolinkml/meta/types/SlotDefinition> .
-          |FILTER NOT EXISTS {
-          |?kid <http://www.w3.org/2000/01/rdf-schema#subPropertyOf>+ ?other .
-          |?other <https://w3id.org/biolink/biolinkml/meta/is_a+/https://w3id.org/biolink/biolinkml/meta/mixins>* ?blslot .
-          |} OPTIONAL { ?kid <http://www.w3.org/2000/01/rdf-schema#label> ?label . }
-          |}""".stripMargin
-      query <- Task.effect(QueryFactory.create(queryText))
-      resultSet <- SPARQLQueryExecutor.runSelectQuery(query)
+      values <- Task.effect(QueryText(predicateMap.zipWithIndex
+        .map(a => String.format(" ( <%s> \"e%s\" ) ", a._1, StringUtils.leftPad(a._2.toString, 4, '0')))
+        .mkString(" ")))
+      queryText = sparql"""SELECT DISTINCT ?qid ?kid ?blslot ?label WHERE { VALUES (?kid ?qid) { $values }
+                     ?kid $rdfSchemaSubPropertyOf+ ?blslot .
+                     ?blslot a $biolinkmlSlotDefinition .
+                     FILTER NOT EXISTS {
+                       ?kid $rdfSchemaSubPropertyOf+ ?other .
+                       ?other <https://w3id.org/biolink/biolinkml/meta/is_a+/https://w3id.org/biolink/biolinkml/meta/mixins>* ?blslot .
```

Comment:
Was this working before? Both the old and new version have a syntax error in:

`<https://w3id.org/biolink/biolinkml/meta/is_a+/https://w3id.org/biolink/biolinkml/meta/mixins>*`

Oh, it's not a syntax error but treats the whole thing as one IRI. This should be a path of two IRIs.

Reply:
Is the following right?

` ?other <https://w3id.org/biolink/biolinkml/meta/is_a>+/<https://w3id.org/biolink/biolinkml/meta/mixins>* ?blslot .`

Reply:
Yes